src/common/PrioritizedQueue.h

```diff
@@ -115,19 +120,35 @@ class PrioritizedQueue {
       else
 	tokens = 0;
     }
-    void enqueue(K cl, unsigned cost, T item) {
+    void enqueue(const K &cl, unsigned cost, const T &item) {
+      int bucket_count = q.bucket_count();
       q[cl].push_back(std::make_pair(cost, item));
-      if (cur == q.end())
+      if (cur == q.end()) {
 	cur = q.begin();
+        cur_key = cur->first;
+      }
+      else if (q.bucket_count() != bucket_count) {
+        cur = q.find(cur_key);
+        assert(cur != q.end());
+        cur_key = cur->first;
```

Comment:
Yeah, so just keep the fail assertion there for confidence?  But at least,  cur_key = cur->first is redundant, I think.